### PR TITLE
Cleanup clipboard interop and add tests

### LINKED ISSUE #1951: Regression: Clipboard.SetText("text", TextDataFormat.Html) adds garbage terminator
Test code from https://github.com/dotnet/winforms/pull/1950

```cs
[StaTheory]
[CommonMemberData(nameof(CommonTestHelper.GetEnumTypeTheoryData), typeof(TextDataFormat))]
public void Clipboard_SetText_InvokeStringTextDataFormat_GetReturnsExpected(TextDataFormat format)
{
    Clipboard.SetText("text", format);
    Assert.Equal("text", Clipboard.GetText(format));
    Assert.True(Clipboard.ContainsText(format));
}
```

Results:
```
xUnit.net Console Runner v2.4.1 (64-bit .NET Core 5.0.0-alpha1.19468.6)
  Discovering: System.Windows.Forms.Tests
  Discovered:  System.Windows.Forms.Tests
  Starting:    System.Windows.Forms.Tests
    System.Windows.Forms.Tests.ClipboardTests.Clipboard_SetText_InvokeStringTextDataFormat_GetReturnsExpected(format: Html) [FAIL]
      Assert.Equal() Failure
                     (pos 4)
      Expected: text
      Actual:   text\0
                     (pos 4)
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs(452,0): at System.Windows.Forms.Tests.ClipboardTests.Clipboard_SetText_InvokeStringTextDataFormat_GetReturnsExpected(TextDataFormat format)
  Finished:    System.Windows.Forms.Tests
=== TEST EXECUTION SUMMARY ===
   System.Windows.Forms.Tests  Total: 12926, Errors: 0, Failed: 1, Skipped: 0, Time: 10.469s
=== COMMAND LINE ===
"D:\a\1\s\.dotnet\dotnet.exe" exec --depsfile "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Release\netcoreapp5.0\System.Windows.Forms.Tests.deps.json" --runtimeconfig "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Release\netcoreapp5.0\System.Windows.Forms.Tests.runtimeconfig.json"  "D:\a\1\s\.packages\xunit.runner.console/2.4.1/tools/netcoreapp2.0/xunit.console.dll" "D:\a\1\s\artifacts\bin\System.Windows.Forms.Tests\Release\netcoreapp5.0\System.Windows.Forms.Tests.dll" -noautoreporters -xml "D:\a\1\s\artifacts\TestResults\Release\System.Windows.Forms.Tests_netcoreapp5.0_x64.xml" -html "D:\a\1\s\artifacts\TestResults\Release\System.Windows.Forms.Tests_netcoreapp5.0_x64.html"  > "D:\a\1\s\artifacts\log\Release\System.Windows.Forms.Tests_netcoreapp5.0_x64.log" 2>&1
```

In .NET Framework the string correctly roundtrips and does not add '\0' at the end

/cc @JeremyKuhne

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup Ole32 Clipboard and HGlobal related interop
- Add Clipboard tests
- Fix #1951

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1950)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Clipboard.cs [hunk 1]
```diff
@@ -2,87 +2,69 @@
...
-        {
-            return IsFormatValid(data.GetFormats());
-        }
-
         internal static bool IsFormatValid(string[] formats)
         {
             Debug.Assert(formats != null, "Null returned from GetFormats");
-            if (formats != null)
+
+            if (formats.Length <= 4)
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/Clipboard.cs]: Are we sure that we won't be null here now? One other thing to consider is changing this to take `ReadOnlySpan<string>` formats.

**Comment by JeremyKuhne** [comment]
How does this fix #1951?

**Comment by hughbe** [comment]
> How does this fix #1951?

Now pushed that commit :)

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+using System.Runtime.InteropServices;
+using System.Threading;
+using Moq;
+using WinForms.Common.Tests;
+using Xunit;
+using IComDataObject = System.Runtime.InteropServices.ComTypes.IDataObject;
+
+namespace System.Windows.Forms.Tests
+{
+    public class ClipboardTests
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ClipboardTests.cs]: Just FYI, there can be concurrency issues with testing static clipboard singleton.
We had discussions some months ago about it - https://github.com/dotnet/winforms/pull/416#issuecomment-463396589

**Comment by RussKie** [comment]
> > How does this fix #1951?
> 
> Now pushed that commit :)

Do we need to patch 3.1?

**Comment by hughbe** [comment]
> Do we need to patch 3.1?

I would say so

**Pull request merged**